src/formatDate.js

```diff
@@ -50,7 +50,51 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat[fromFormat.length - 1];
+  const separatorTo = toFormat[toFormat.length - 1];
+  const dateArray = date.split(separatorFrom);
+  const result = [];
+  let day;
+  let month;
+  let year;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    switch (fromFormat[i][0]) {
+      case 'Y':
+        year = dateArray[i];
+        break;
+      case 'M':
+        month = dateArray[i];
+        break;
+      case 'D':
+        day = dateArray[i];
+        break;
+    }
+  }
+
+  for (let k = 0; k < toFormat.length; k++) {
+    switch (toFormat[k][0]) {
+      case 'D':
+        result.push(day);
+        break;
+      case 'Y':
+        if (toFormat[k] === 'YYYY' && year.length <= 2 && +year < 30) {
+          result.push('20' + year);
+        } else if (toFormat[k] === 'YYYY' && year.length <= 2) {
+          result.push('19' + year);
+        } else if (toFormat[k] === 'YY' && year.length > 2) {
+          result.push(year.slice(2));
+        } else {
+          result.push(year);
+        }
```

Comment:
For all of these conditions, the best way use constants. Also, in this case, I see magic numbers. What does mean 2 in this case? What mean 30? Add some readability for this part)
Thanks!